survey/views.py

```diff
@@ -20,37 +22,95 @@
 
 from home.models import Project
 from survey.services import survey_service
+from .forms import InvitationForm
+from .models import Survey
+import logging
 
 from .forms import InvitationForm
 from .models import Survey, SurveyEvidenceSection, SurveyEvidenceFile
 from .services.survey import InvalidInviteTokenException
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseForbidden
+from .services.survey import SurveyService
+from survey.dashboards.dashboard import get_survey_dashboard
 
 logger = logging.getLogger(__name__)
 
 
 class SurveyView(LoginRequiredMixin, View):
+    login_url = '/login/'
     """
     Manager's view of a survey to be sent out. The manager is able to
     configure what fields are included in the survey on this page.
     """
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.survey_service = SurveyService()
```

Comment:
This doesn't feel right. The survey service is initialised in the `survey.services` module already. I think should be in the global scope like this instead, it's at the service "layer" of the app:

```python
from .services import survey_service
```
There are references to this `survey_service` global variable in this view, so I don't think this code will work as it is?